snowpack/src/build/build-pipeline.ts

```diff
@@ -140,17 +156,22 @@ async function runPipelineTransformStep(
           urlPath: `./${path.basename(rootFileName + destExt)}`,
         });
         logger.debug(`✔ transform() success [${debugPath}]`, {name: step.name});
-        // if step returned a value, only update code (don’t touch .map)
         if (typeof result === 'string' || Buffer.isBuffer(result)) {
           output[destExt].code = result;
           output[destExt].map = undefined;
         } else if (result && typeof result === 'object' && (result as {result: string}).result) {
           output[destExt].code = (result as {result: string}).result;
-          output[destExt].map = undefined;
+          let map = (result as {map: string | RawSourceMap}).map;
+          let outputMap: string | undefined = undefined;
+          if (map && sourceMaps) {  // if source maps disabled, don’t return any
+            if (output[destExt].map) {
+              outputMap = await composeSourceMaps(filePath, output[destExt].map!, map);
+            } else {
+              outputMap = typeof map === 'object' ? JSON.stringify(map) : map;
+            }
+          }
+          output[destExt].map = outputMap;
```

Comment:
I think if source maps are disabled, we still want to unset `output[destExt].map = undefined`, otherwise Snowpack will write that sourcemap to disk

Reply:
Also removing in favor of below changes that’d affect this.